--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -302,16 +302,18 @@ set(WASM_DEFAULT_LDC_SWITCHES "${WASM_DEFAULT_LDC_SWITCHES}\n        \"-L--expor
 set(LDC_BUILD_IMPORT_DIR "${PROJECT_BINARY_DIR}/import")
 
 # Only generate the config files if this CMake project is embedded in the LDC CMake project.
-if(LDC_EXE)
+if(LDC_EXE OR CONFIG_FILE_OUTPUT_DIR)
+    set(LDC_EXE "ldc2")
     if(PHOBOS2_DIR)
         set(CONFIG_NAME ${LDC_EXE}_phobos)
     else()
         set(CONFIG_NAME ${LDC_EXE})
     endif()
-    set(conf_path         ${PROJECT_BINARY_DIR}/../bin/${LDC_EXE}.conf)
-    set(install_conf_path ${PROJECT_BINARY_DIR}/../bin/${LDC_EXE}_install.conf)
-    configure_file(${PROJECT_PARENT_DIR}/${CONFIG_NAME}.conf.in ${conf_path} @ONLY)
-    configure_file(${PROJECT_PARENT_DIR}/${LDC_EXE}_install.conf.in ${install_conf_path} @ONLY)
+    set(conf_path         ${CONFIG_FILE_OUTPUT_DIR}/${LDC_EXE}.conf)
+    set(install_conf_path ${CONFIG_FILE_OUTPUT_DIR}/${LDC_EXE}_install.conf)
+    configure_file(${PROJECT_SOURCE_DIR}/../${CONFIG_NAME}.conf.in ${conf_path} @ONLY)
+    configure_file(${PROJECT_SOURCE_DIR}/../${LDC_EXE}_install.conf.in ${install_conf_path} @ONLY)
+    unset(LDC_EXE)
 
     # macOS has fat libraries; otherwise, append a separate config file section for the
     # multilib target and override the lib directory.
